File: app/src/main/java/com/commonpepper/photosen/ui/viewmodels/PhotoFragmentViewModelFactory.java
package com.commonpepper.photosen.ui.viewmodels;

import com.commonpepper.photosen.network.datasource.PhotoListDataSourceFactory;
import com.commonpepper.photosen.model.Photo;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class PhotoFragmentViewModelFactory implements ViewModelProvider.Factory {

    private String date;

    public PhotoFragmentViewModelFactory(String date) {
        this.date = date;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new PhotoListFragmentViewModel();
    }

    public class PhotoListFragmentViewModel extends AbstractListFragmentViewModel<Photo> {

        @Override
        public PhotoListDataSourceFactory createDataSourceFactory() {
            return new PhotoListDataSourceFactory(date);
        }
    }

}

File: app/src/main/java/com/commonpepper/photosen/ui/viewmodels/PhotoFragmentViewModelFactory.kt
package com.commonpepper.photosen.ui.viewmodels

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.Factory
import com.commonpepper.photosen.model.Photo
import com.commonpepper.photosen.network.datasource.PhotoListDataSourceFactory

class PhotoFragmentViewModelFactory(private val date: String) : Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return PhotoListFragmentViewModel() as T
    }

    inner class PhotoListFragmentViewModel : AbstractListFragmentViewModel<Photo?>() {
        override fun createDataSourceFactory(): PhotoListDataSourceFactory {
            return PhotoListDataSourceFactory(date)
        }
    }

}